File: Framework/LanguageSupport/cpp/MMICPP/Adapter/SessionData.cpp
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Andreas Kaiser, Niclas Delfs, Stephan Adam

#include "SessionData.h"

using namespace MMIStandard;

//initialize static members
MAdapterDescription SessionData::adapterDescription;
MIPAddress SessionData::registerAddress;
std::vector<MMUDescription> SessionData::mmuDescriptions;
std::unordered_map<std::string, std::string> SessionData::mmuPaths;
time_t SessionData::startTime;
time_t SessionData::lastAccess=0;
Concurrency::concurrent_unordered_map<std::string, unique_ptr<SessionContent>> SessionData::SessionContents;
 

const MMUDescription &SessionData::GetMMUDescription(string mmuId)
{
	for (const MMUDescription &desc : mmuDescriptions)
	{
		if (desc.ID.compare(mmuId)==0)
			return desc;
	}
	throw runtime_error("No MMUDescription with id:" + mmuId + " found");
}

const MIPAddress & SessionData::GetRegisterAddress()
{ 
	return  registerAddress;
}

File: Framework/LanguageSupport/java/MMIJava/src/Adapter/SessionData.java
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Andreas Kaiser

package Adapter;


import de.mosim.mmi.core.MIPAddress;
import de.mosim.mmi.mmu.MMUDescription;
import de.mosim.mmi.register.MAdapterDescription;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.ConcurrentMap;

public class SessionData {
    /*
		Class which contains the data of the sessions and MMUs
	*/

    //	The description of the adapter
    public static MAdapterDescription adapterDescription;
    //	The address of the MMIRegister
    public static MIPAddress registerAddress;
    //	The last time the adapter was used
    public static long lastAccess = 0;
    //	The time when the adapter was started
    public static long startTime;
    //	Map which contains all sessions
    public static ConcurrentMap<String, SessionContent> sessionContents = new ConcurrentHashMap<>();
    //  MMUID to MMUDescription mapping
    public static Map<String, MMUDescription> MMUZipEntry = new HashMap<>();
    //	Descriptions of the loadable MMUs>
    public static List<MMUDescription> MMUDescriptions = new ArrayList<>();
}
